ambry-store/src/main/java/com/github/ambry/store/StorageManager.java

```diff
@@ -551,6 +564,11 @@ public void onPartitionBecomeBootstrapFromOffline(String partitionName) {
               "Store " + partitionName + " didn't start correctly, replica should be set to ERROR state",
               StoreNotStarted);
         }
+
+        if (!isPrimaryHelixListener) {
```

Comment:
Let me take another look

Reply:
The purpose of listening on both helix servers is to make sure when one of the helix transitioned the state to ERROR, the second one would do the same. 

There are several cases in this PR, when we terminate early for the second helix, it will not ends up in ERROR state. We should let the code in the those callback executed for both helix.

Reply:
Where is this callback below the return statement ?
In both cases where it returns early, there is no callback below.
The code below the return just creates some files (or not) and sets/resets some state.